components/library/src/library.rs

```diff
@@ -51,22 +57,78 @@ impl Library {
             paths_to_pages: HashMap::with_capacity(cap_pages),
             paths_to_sections: HashMap::with_capacity(cap_sections),
             is_multilingual,
+            reverse_aliases: HashMap::new(),
+            translations: HashMap::new(),
         }
     }
 
     /// Add a section and return its Key
     pub fn insert_section(&mut self, section: Section) -> DefaultKey {
-        let path = section.file.path.clone();
+        let file_path = section.file.path.clone();
+        let file_rel_path = section.file.relative.clone();
+        let rel_path = section.path.clone();
+
+        self.reverse_aliases
+            .entry(rel_path.clone())
+            .and_modify(|s| {
+                s.insert(file_rel_path.clone());
+            })
+            .or_insert_with(|| {
+                let mut s = HashSet::new();
+                s.insert(file_rel_path.clone());
+                s
+            });
+
+        for alias in &section.meta.aliases {
+            self.reverse_aliases
+                .entry(alias.to_string())
+                .and_modify(|s| {
+                    s.insert(file_rel_path.clone());
+                })
+                .or_insert_with(|| {
+                    let mut s = HashSet::new();
+                    s.insert(file_rel_path.clone());
+                    s
```

Comment:
Not quite sure how that would work, as l 71-80 isn't actually a loop, it just creates the "identity alias". We do need the loop though because the aliases are the keys with the same values. So I'm not sure how you would do that.

Reply:
Done